src/Framework/Migrations/MigrationsRunner.php

```diff
@@ -103,19 +97,55 @@ public function run()
 
             $migrationLog = $this->migrationLogFactory->make($migrationId);
 
-            // Begin transaction
-            $wpdb->query('START TRANSACTION');
-
             try {
-                /** @var Migration $migration */
+                /**
+                 * @var Migration|BatchMigration $migration
+                 */
                 $migration = give($migrationClass);
 
-                $migration->run();
-
-                // Save migration status
-                $migrationLog->setStatus(MigrationLogStatus::SUCCESS);
+                if ($migration instanceof BatchMigration) {
+                    $status = (new BatchMigrationRunner($migration))->run();
+
+                    if ($status === MigrationLogStatus::RUNNING) {
+                        give()->notices->register_notice(
+                            [
+                                'id' => $migrationId,
+                                'description' => sprintf(
+                                    esc_html__('Running DB migration: %s', 'give'),
+                                    $migration::title()
+                                ),
+                            ]
+                        );
+
+                        // Update status to RUNNING
+                        if (MigrationLogStatus::RUNNING !== $migrationLog->getStatus()) {
+                            $migrationLog->setStatus(MigrationLogStatus::RUNNING);
+                            $migrationLog->save();
+                        }
+
+                        break;
+                    }
+
+                    if ($status === MigrationLogStatus::INCOMPLETE) {
+                        give()->notices->register_notice(
+                            [
+                                'id' => $migrationId,
+                                'type' => 'warning',
+                                'description' => sprintf(
+                                    esc_html__('Incomplete DB migration: %s', 'give'),
```

Comment:
What does this mean for the admin? What should they do at this point?

Reply:
They should go to the migrations list table where they can re-run the failed migrations. We should probably add a link to the migrations list table here.

Reply:
Yup, then let's update this description to provide instructions.

Reply:
Resolved https://github.com/impress-org/givewp/pull/7755/commits/057ea159288bc94947079c914038841690545a3b